packaging/tags.py

```diff
@@ -101,7 +101,8 @@ def __eq__(self, other: object) -> bool:
             return NotImplemented
 
         return (
-            (self.platform == other.platform)
+            (self._hash == other._hash)
+            and (self.platform == other.platform)
```

Comment:
FWIW, we'd also benefit from comparing against _platform (and other private names) instead of properties, reducing another level of indirection here.

Reply:
I profiled that change locally, and the difference is "in the noise".  Given that and the conversation in #179 (about which I don't really have context) I left it as is.  But happy to push that change if you'd like.

Reply:
If we are going to bother to optimize this then I say let's go all the way and use the private attributes.